app/uk/gov/hmrc/hec/repos/HECTaxCheckStore.scala

```diff
@@ -75,8 +77,10 @@ class HECTaxCheckStoreImpl @Inject() (
   private val isExtractedField: String = s"data.$key.isExtracted"
 
   private val hecIndexes = Seq(
-    Index(Seq(ggCredIdField -> IndexType.Ascending)),
-    Index(Seq(isExtractedField -> IndexType.Ascending))
+    Index(
+      Seq(ggCredIdField -> IndexType.Ascending),
+      partialFilter = Some(BSONDocument(isExtractedField -> BSONDocument(f"$$exists" -> false)))
```

Comment:
what's the requirement for `$$exists`? wouldn't `Some(BSONDocument(isExtractedField -> false))` suffice?

Reply:
i was doing it for first time , so checked on internet. I think that was for an optional field. osExtracted is not optional, so we can remove that exists

Reply:
[updated](https://github.com/hmrc/hec/pull/21/commits/c214831ee5fe80310451c4d80ea60982d72db436)

Reply:
cool, I'm a bit unsure about how the indexes are getting created here, I think it should be separate indexes like below, but not at all sure, @AndyHWChung are there examples of usage of `partialFilter` in another repo?

you can always run it and see how the indexes have been created in the mongo database using `db.hecTaxChecks.getIndexes()`
```
private val hecIndexes = Seq(
    Index(Seq(ggCredIdField -> IndexType.Ascending)),
    Index(
      Seq("isExtractedField" -> IndexType.Ascending),
      partialFilter = Some(BSONDocument("isExtractedField" -> false))
    )
  )
```

Reply:
I'm haven't seen it before, but I could do some digging around. Doing a github search on `partialFilter` on the whole HMRC organisation should tell us

Reply:
found this using the approach I just described:
https://github.com/hmrc/agent-mapping/blob/f383cd37e69cdb995a6ad0b41406f5233b021d7c/app/uk/gov/hmrc/agentmapping/repository/MappingRepository.scala#L126
that's just the first concrete thing I came across though so there could be other examples as well

Reply:
i can try this one and see

Reply:
i tested with 
`private val hecIndexes = Seq(
    Index(Seq(ggCredIdField -> IndexType.Ascending)),
    Index(
      Seq("isExtractedField" -> IndexType.Ascending),
      partialFilter = Some(BSONDocument("isExtractedField" -> false))
    )
  )`
  
  i can see only two indexes one on _id and one on modifiedDetails.lastUpdated

Reply:
updated